### PR TITLE
UX DoD Requirements

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add UX DoD requirements document

### CODE REVIEW

File: UX_DOD_REQUIREMENTS.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+## General information
+
+UI developed within the Console and Compass UI should be compliant with [Fiori Design Guidelines](https://experience.sap.com/fiori-design-web/).
+
+## Additional guidelines for modals
+
+* User should be able to close the modal (without saving changes, if any) by clicking anywhere outside of it.
+* Modal should support keyboard control:
+    * If modal contains input fields, first of them should be automatically focused on opening the modal. In some cases, we can focus the most important element insted (e.g. invalid one).
```
Comment (dariadomagala) [inline UX_DOD_REQUIREMENTS.md]:
```suggestion
    * If modal contains input fields, first of them should be automatically focused on opening the modal. In some cases, we can focus the most important element instead (e.g. invalid one).
```

File: UX_DOD_REQUIREMENTS.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    * If there is no input fields, user still should be able to navigate the modal by the means of keyboard only. Until modal is opened, focus should be trapped inside.
```
Comment (dariadomagala) [inline UX_DOD_REQUIREMENTS.md]: I don't think I understand what you meant here.

File: UX_DOD_REQUIREMENTS.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    * User should be able to close the modal by hitting the escape key (equivalent to clicking the `X` or `Cancel` button).
+    * User should be able to confirm introduced changes (if any) by pressing the `enter`/`return` key (equivalent to clicking the `Confirm` button).
```
Comment (dariadomagala) [inline UX_DOD_REQUIREMENTS.md]: Should he? We have the 'labels' input that adds a new label by pressing the 'enter' button.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,4 +1,25 @@
...
+
+### General information
+
+UI developed within the Console and Compass UI should be compliant with [Fiori Design Guidelines](https://experience.sap.com/fiori-design-web/).
+
+### Additional guidelines for modals
+
+* User should be able to close the modal (without saving changes, if any) by clicking anywhere outside of it.
+* Modal should support keyboard control:
+    * User should be always able to navigate the modal by the means of keyboard only. While the modal is opened, focus should not leave the modal window.
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
    * User should always be able to navigate the modal using only a keyboard. While the modal is opened, focus should not leave the modal window.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,4 +1,25 @@
...
+    * If modal contains input fields, first of them should be automatically focused on opening the modal. In some cases, we can focus the most important element insted (e.g. invalid one).
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
    * If modal contains input fields, the first of them should be automatically focused on opening the modal. In some cases, we can focus on the most important element instead.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,4 +1,25 @@
...
+    * Main action (e.g. _Confirm_, _Save_, _Apply_) should use _emphasized_ style (read more [here](https://experience.sap.com/fiori-design-web/button)).
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
    * Main actions, such as **Confirm**,  **Save**, **Apply**, should use the [emphasized style](https://experience.sap.com/fiori-design-web/button).
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,4 +1,25 @@
...
+* If content of the modal is dynamic, try to avoid resizing the modal itself. What's more, modals' widths should be consistent across the application.
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
* If the content of the modal is dynamic, avoid resizing the modal. Modals width should be consistent across the application.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,4 +1,25 @@
...
+* Remember about [ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) attributes.
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
* Follow the [ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) attributes.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,4 +1,25 @@
...
+
+### Additional guidelines for Compass
+
+* While creating validable forms, make sure to meet the requirements outlined [here](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-input-validation.md).
+* While using HTML input `type` attribute for validation, make sure to verify that their validation corresponds to one used on backend.
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
* While using HTML input `type` attribute for validation, make sure to verify if their validation corresponds to the one used on the backend.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,4 +1,25 @@
...
+### General information
+
+UI developed within the Console and Compass UI should be compliant with [Fiori Design Guidelines](https://experience.sap.com/fiori-design-web/).
+
+### Additional guidelines for modals
+
+* User should be able to close the modal (without saving changes, if any) by clicking anywhere outside of it.
+* Modal should support keyboard control:
+    * User should always be able to navigate the modal using only a keyboard. While the modal is opened, focus should not leave the modal window.
+    * If a modal contains input fields, the first of them should be automatically focused on opening the modal. In some cases, we can set focus on the most important element instead (e.g. incorrect one).
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
    * If the modal contains input fields, the first of them should be automatically focused on opening the modal. In some cases, we can set focus on the most important element instead (e.g. the incorrect one).
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,4 +1,25 @@
...
+    * Main actions, such as **Confirm**,  **Save**, **Apply**, should use the [emphasized style](https://experience.sap.com/fiori-design-web/button).
+* If the content of the modal is dynamic, avoid resizing the modal. Modals width should be consistent across the application.
+* Follow the [ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) attributes.
+
+### Additional guidelines for Compass
+
+* While creating validable forms, make sure to meet the requirements outlined [here](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-input-validation.md).
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
* When creating validable forms, make sure to meet the requirements outlined [here](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-input-validation.md).
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,4 +1,25 @@
...
+* While using HTML input `type` attribute for validation, make sure to verify if their validation corresponds to the one used on the backend.
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]:
```suggestion
* When using HTML input `type` attribute for validation, make sure to verify if the validation corresponds to the one used on the backend.
```

**Pull request merged**